### DISCUSSION TITLE
Various Questions

### DISCUSSION THREAD

**Discussion by Chrispapge96** [IBM/aihwkit#478] [Q&A] [answered]
Hello, I have a couple of questions regarding the simulator.

1.Is the VectorUnitCell adding the devices in series? Can I make them be in parallel? 
2.Is it possible to have synaptic or input bit slicing in the simulator?
3.Is it possible to program the weights with different voltage pulse?
4.How do weight scaling and mapping work? Could you refer me to some reading?
5.I am also looking on how to set my IOParameters and I would gladly accept any suggestions of studies regarding
noise sources in a tile, (apart from device level non-idealities).

Thank you in advance!

**Comment by maljoras** [comment, marked as answer]
Hi @Chrispapge96:
1) The devices of the `VectorDevice` are implementing something like `y = alpha * W_1 x + beta * W_2 x` where the summation is thought to be in currents (not in digital after going through the ADC).  The significance factors can be set. 
2) Input slicing is not supported, however, it would be easy to implement by modifying the input before passing to the analog tile with standard torch syntax.
3) There are some selectors to program the weights in the VectorDevice differently, for instance a random selection or all simultaneously.  Other forms are no implemented so far.  However, you could build your "Vector device" by adding up several AnalogLinear modules, with the caveat that then the output addition would be done in "digital" 
4) The weight scaling simply scales the max value of the given weight before setting it to the devices and adjust the digital output factors accordingly.  This can we done column-wise also, where one assumes a digital output scaling for each column. This typically increases the usable range of the conductances. See also the paper below where this is explained in detail (in an analog inference context)
5) While we do not have an exact hardware fit at the moment, you can take a look at the study https://arxiv.org/abs/2302.08469 where some reasonable assumptions are shown in the appendix.

**Reply by Chrispapge96** [reply]
That is great! Exactly what I was looking for! Thank you very much
